modules/gateway/gateway.go

```diff
@@ -159,18 +144,26 @@ func New(addr string, persistDir string) (g *Gateway, err error) {
 	g.log.Println("INFO: gateway created, started logging")
 
 	// Forward the RPC port, if possible.
-	go g.forwardPort(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
+		g.forwardPort(port)
```

Comment:
I've been using the `threaded` prefix to indicate that the method needs to be called from a goroutine, not that it _should_ be called from a goroutine. Same goes for `learnHostname`.